File: frontend/src/types/trade.ts
export type TradeType = 'BUY' | 'ADD_BUY' | 'SELL' | 'PARTIAL_SELL';

export interface Trade {
  id: string;
  position_id: string;
  trade_type: TradeType;
  trade_date: string;
  price: number;
  quantity: number;
  total_amount: number;
  realized_profit?: number;
  realized_return_pct?: number;
  avg_price_after?: number;
  quantity_after?: number;
  reason?: string;
  notes?: string;
  stock_code?: string;
  stock_name?: string;
  created_at: string;
}

export interface TradeUpdateData {
  price?: number;
  quantity?: number;
  trade_date?: string;
  reason?: string;
  notes?: string;
}

export interface TradeListResponse {
  trades: Trade[];
  total_count: number;
}

export interface TradeSummary {
  total_trades: number;
  buy_count: number;
  sell_count: number;
  total_buy_amount: number;
  total_sell_amount: number;
  total_realized_profit: number;
  winning_trades: number;
  losing_trades: number;
  win_rate: number;
  avg_profit_per_trade: number;
  avg_return_pct: number;
}

export interface MonthlyTradeStats {
  month: string;
  trade_count: number;
  buy_count: number;
  sell_count: number;
  realized_profit: number;
  win_rate: number;
}

export interface TickerTradeStats {
  ticker: string;
  stock_name?: string;
  trade_count: number;
  total_buy_amount: number;
  total_sell_amount: number;
  realized_profit: number;
  avg_return_pct: number;
  winning_trades: number;
  losing_trades: number;
  win_rate: number;
}

export interface TradeAnalysisResponse {
  summary: TradeSummary;
  monthly_stats: MonthlyTradeStats[];
  ticker_stats: TickerTradeStats[];
  recent_trades: Trade[];
}

File: backend/models/trade.py
"""매매 기록 모델."""
import uuid
from datetime import datetime, date
from enum import Enum as PyEnum

from sqlalchemy import Column, String, Integer, Numeric, Date, DateTime, ForeignKey, Text, Enum
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship

from core.database import Base


class TradeType(str, PyEnum):
    BUY = "BUY"
    ADD_BUY = "ADD_BUY"
    SELL = "SELL"
    PARTIAL_SELL = "PARTIAL_SELL"


class Trade(Base):
    __tablename__ = "trades"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    position_id = Column(UUID(as_uuid=True), ForeignKey("positions.id"), nullable=False)

    trade_type = Column(Enum(TradeType), nullable=False)
    trade_date = Column(Date, default=date.today, nullable=False)
    price = Column(Numeric(15, 2), nullable=False)
    quantity = Column(Integer, nullable=False)

    # 매도 시 실현손익
    realized_profit = Column(Numeric(15, 2), nullable=True)
    realized_return_pct = Column(Numeric(10, 2), nullable=True)

    # 거래 후 포지션 상태
    avg_price_after = Column(Numeric(15, 2), nullable=True)
    quantity_after = Column(Integer, nullable=True)

    reason = Column(String(200), nullable=True)
    notes = Column(Text, nullable=True)

    # 종목 정보 (조회 편의용)
    stock_code = Column(String(20), nullable=True)
    stock_name = Column(String(100), nullable=True)

    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)

    position = relationship("Position", backref="trades")
